nucypher/characters/lawful.py

```diff
@@ -886,11 +874,13 @@ def retrieve(self,
 
         for message_kit in message_kits:
             message_kit.set_correctness_keys(receiving=self.public_keys(DecryptingPower))
-            message_kit.set_correctness_keys(verifying=alice_verifying_key)
+            message_kit.set_correctness_keys(verifying=relayer_verifying_key)
```

Comment:
Should be Alice's key here

Reply:
Help me clarify here: We expect that the KFrag generator constructed this MessageKit, and not the publisher?

Reply:
The verifying key here is the verifying key that was used to sign kfrags, so it should be the Alice's key. It's actually not related to the creator of the message kit at all (and we don't even serialize it with the kit); currently `UmbralMessageKit` only carries kfrag verification keys for convenience.